### PR TITLE
feat: add SonarQube image and stage template

### PR DESCRIPTION
**Pull request by cd1989**
**What this PR does / why we need it**:

- Support SonarQube by stage template
- Create image for SonarQube scan

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @zhujian7 @supereagle 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: manifests/templates/cicd/sonarqube.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
+- apiVersion: cyclone.dev/v1alpha1
+  kind: Stage
+  metadata:
+    name: sonarqube-template
```
Comment (supereagle) [inline manifests/templates/cicd/sonarqube.yaml]: codescan-template?

File: build/cicd/sonarqube/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FROM newtmitch/sonar-scanner:3.3.0-alpine
+
+LABEL maintainer="chende@caicloud.io"
+
+ENV WORKDIR /workspace
+WORKDIR $WORKDIR
+
+RUN wget -O /bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 && chmod +x /bin/jq
```
Comment (supereagle) [inline build/cicd/sonarqube/Dockerfile]:
```suggestion
RUN wget -O /bin/jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 && \
         chmod +x /bin/jq
```

File: build/cicd/sonarqube/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+COPY ./build/cicd/sonarqube/entrypoint.sh /
+
+ENTRYPOINT ["/entrypoint.sh"]
```
Comment (supereagle) [inline build/cicd/sonarqube/Dockerfile]: Add new line

File: manifests/templates/cicd/sonarqube.yaml (hunk 1 above)
Comment (cd1989) [reply manifests/templates/cicd/sonarqube.yaml]: I'd prefer sonarqube-template, it's more accurate.

**Pull request merged**